src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationServiceRunner.cs

```diff
@@ -99,43 +99,54 @@ private ImmutableArray<ITypeSymbol> GetTypeSymbols(ImmutableArray<ISymbol> candi
             {
                 if (candidateSymbol is IMethodSymbol method)
                 {
-                    if (method.Parameters.Length > ordinalInInvocation)
+                    ITypeSymbol type;
+                    if (method.IsParams() && (ordinalInInvocation >= method.Parameters.Length - 1))
                     {
-                        var type = method.Parameters[ordinalInInvocation].Type;
-                        // If type is <see cref="Expression{TDelegate}"/>, ignore <see cref="Expression"/> and use TDelegate.
-                        // Ignore this check if expressionSymbol is null, e.g. semantic model is broken or incomplete or if the framework does not contain <see cref="Expression"/>.
-                        if (expressionSymbol != null &&
-                            type is INamedTypeSymbol expressionSymbolNamedTypeCandidate &&
-                            expressionSymbolNamedTypeCandidate.OriginalDefinition.Equals(expressionSymbol))
+                        var arrayType = method.Parameters.LastOrDefault().Type;
+                        type = ((IArrayTypeSymbol)arrayType).ElementType;
```

Comment:
i don't know if it's guaranteed that it will be an array.  can you test something like `params int i`?

Reply:
also, the compiler/language may not always provide an array here in the future (for example: params IEnumerable).  So best to actually explicitly check for the array case and bail otherwise.

Reply:
param int i is not an allowed syntax: The params parameter must be a single dimensional array.

I agree that we should check if it is an array and bail out otherwise. However, I do not think we should have a test case for the incorrect syntax.

---
In reply to: [289223266](https://github.com/dotnet/roslyn/pull/36040#discussion_r289223266) [](ancestors = 289223266)

Reply:
> param int i is not an allowed syntax: The params parameter must be a single dimensional array.

I don't think that's strictly true.  Namely, the *parser* totally allows this (though later passes may error if they see somethin that was a non-array).

> I agree that we should check if it is an array and bail out otherwise. However, I do not think we should have a test case for the incorrect syntax.

The test case just validates we dont crash :)  I don't expect the user experience to be one we cater to.  I just want us to not crash, and to have a test that prevents us from regressing in the future :)